k2/csrc/util.h

```diff
@@ -7,15 +7,25 @@
 #ifndef K2_CSRC_UTIL_H_
 #define K2_CSRC_UTIL_H_
 
+#include <algorithm>
 #include <cfloat>
 #include <cmath>
 #include <functional>
+#include <limits>
 #include <utility>
 
 #include "k2/csrc/fsa.h"
 
 namespace k2 {
 
+#define EXPECT_DOUBLE_ARRAY_APPROX_EQ(expected, actual, abs_error)             \
+  EXPECT_EQ(expected.size(), actual.size()) << "Different Array Size.";        \
+  for (std::size_t i = 0; i < std::min(expected.size(), actual.size()); ++i) { \
+    EXPECT_TRUE(ApproxEqual(expected[i], actual[i], abs_error))                \
```

Comment:
There is a macro `EXPECT_NEAR` which does the same thing as this line, but much simpler.

Please refer to https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#floating-point-macros

Reply:
it's sad that macro does not work for `infinity`, I have tried before...